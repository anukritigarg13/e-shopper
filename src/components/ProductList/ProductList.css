.product-list{
    display:flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    flex-grow: 1;
}
.product-type{
    margin: 0% 5%;
}
.product-container{
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 2% 5%;
}
@media screen and (max-width:912px){
    .product-list{
        display:flex;
        justify-content: center;
        flex-wrap: wrap;
    }
    .product-type{
      margin: 5% 5%;
      padding: 0% 5%;
    }
}