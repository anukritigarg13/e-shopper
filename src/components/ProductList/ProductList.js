import React from 'react';
import PropTypes from 'prop-types';
import Product from '../Product/Product';
import './ProductList.css';

const ProductList = ({
  products, add, remove, category,
}) => {
  const requiredProducts = products.map((product) => (
    // <Product
    //   key={product.id}
    //   itemName={product.itemName}
    //   imgSrc={product.imgSrc}
    //   imgAlt={product.imgAlt}
    //   companyName={product.companyName}
    //   unitPrice={product.unitPrice}
    //   unitQuantity={product.unitQuantity}
    //   itemCount={product.itemCount}
    //   add={() => add(product.id)}
    //   remove={() => remove(product.id)}
    // />
    <Product
      key={product.id}
      itemName={product.name}
      imgSrc="assets/images/grapes.jpeg"
      imgAlt="image of grapes"
      companyName="Fresco"
      unitPrice={product.price}
      unitQuantity="1 kg"
      stock={product.count}
      itemCount={product.itemCount}
      add={() => add(product.id, product.category)}
      remove={() => remove(product.id, product.category)}
      category={product.category}
    />
  ));
  return (
    <div className="product-container">
      <div className="product-type">{category}</div>
      <div className="product-list">
        {requiredProducts}
      </div>
    </div>
  );
};
const productsShape = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  itemCount: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired,
};
ProductList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape(productsShape)).isRequired,
  add: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
  category: PropTypes.string.isRequired,
};
export default ProductList;
