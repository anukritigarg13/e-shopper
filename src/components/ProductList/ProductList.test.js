/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import ProductList from './ProductList';

describe('ProdcutList component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  let mockProps;
  beforeAll(() => {
    mockProps = {
      add: () => {},
      remove: () => {},
      category: 'Fruits',
      products: [{
        id: 1,
        name: 'MOCK_ITEM_NAME_1',
        itemCount: 3,
        count: 2,
        price: 40,
        add: () => {},
        remove: () => {},
        category: 'Fruits',
      }, {
        id: 2,
        name: 'MOCK_ITEM_NAME_2',
        itemCount: 3,
        count: 2,
        price: 40,
        add: () => {},
        remove: () => {},
        category: 'Fruits',
      }],
    };
  });
  it('should render two Product components', () => {
    render(<ProductList {...mockProps} />);
    const prodcutElements = screen.getAllByTestId('product-test');
    expect(prodcutElements.length).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<ProductList {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
