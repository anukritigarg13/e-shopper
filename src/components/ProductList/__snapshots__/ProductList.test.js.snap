// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProdcutList component should match snapshot 1`] = `
<div>
  <div
    class="product-container"
  >
    <div
      class="product-type"
    >
      Fruits
    </div>
    <div
      class="product-list"
    >
      <div
        class="product"
        data-testid="product-test"
      >
        <div
          class="product-image-div"
        >
          <img
            alt="image of grapes"
            class="product-image"
            data-testid="product-img"
            src="assets/images/grapes.jpeg"
          />
        </div>
        <p>
          Fresco
        </p>
        <p>
          MOCK_ITEM_NAME_1
        </p>
        <p>
          1 kg
        </p>
        <div
          class="quantity-placard"
        >
          <p>
            MRP
             
            40
            /-
          </p>
          <div
            class="quantity-control"
            data-testid="quantity-control"
          >
            <p>
              <button
                class="control-button"
                type="button"
              >
                +
              </button>
            </p>
            <p
              class="quantity-in-basket"
            >
              3
               
              in Basket
            </p>
            <p>
              <button
                class="control-button"
                type="button"
              >
                -
              </button>
            </p>
          </div>
        </div>
      </div>
      <div
        class="product"
        data-testid="product-test"
      >
        <div
          class="product-image-div"
        >
          <img
            alt="image of grapes"
            class="product-image"
            data-testid="product-img"
            src="assets/images/grapes.jpeg"
          />
        </div>
        <p>
          Fresco
        </p>
        <p>
          MOCK_ITEM_NAME_2
        </p>
        <p>
          1 kg
        </p>
        <div
          class="quantity-placard"
        >
          <p>
            MRP
             
            40
            /-
          </p>
          <div
            class="quantity-control"
            data-testid="quantity-control"
          >
            <p>
              <button
                class="control-button"
                type="button"
              >
                +
              </button>
            </p>
            <p
              class="quantity-in-basket"
            >
              3
               
              in Basket
            </p>
            <p>
              <button
                class="control-button"
                type="button"
              >
                -
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
